rust/src/builders/witness_builder.rs

```diff
@@ -303,6 +303,54 @@ impl TransactionWitnessSetBuilder {
         };
     }
 
+    fn add_vkeys(&mut self, vkeys: &Vec<Vkey>) {
+        let fake_sig = fake_raw_key_sig(0);
+        for vkey in vkeys {
+            let fake_vkey_witness = Vkeywitness::new(&vkey, &fake_sig);
+            self.add_vkey(&fake_vkey_witness);
+        }
+    }
+
+    fn add_vkeys_by_num(&mut self, num: usize) {
+        let vkeys: Vec<Vkey> = (0..num).into_iter().map(|_| Vkey::new(&fake_raw_key_public(0))).collect();
+        self.add_vkeys(&vkeys);
+    }
+
+    pub fn add_input_aggregate_witness_data(&mut self, data: &InputAggregateWitnessData) {
+        match data {
+            InputAggregateWitnessData::Vkeys(vkeys) => self.add_vkeys(vkeys),
+            InputAggregateWitnessData::Bootstraps(witnesseses) => {
+                for witness in witnesseses {
+                    self.add_bootstrap(witness);
+                }
+            }
+            InputAggregateWitnessData::NativeScript(script, info) => {
+                self.add_native_script(script);
+                match info.0 {
+                    NativeScriptWitnessInfoKind::Count(num) => self.add_vkeys_by_num(num),
+                    NativeScriptWitnessInfoKind::Vkeys(ref vkeys) => self.add_vkeys(vkeys),
+                    NativeScriptWitnessInfoKind::AssumeWorst => {
+                        let num = script.get_required_signers().len();
+                        self.add_vkeys_by_num(num);
+                    }
+                }
+            }
+            InputAggregateWitnessData::PlutusScriptNoDatum(witness, info) => {
```

Comment:
I'm not sure about the Plutus part. Am I doing right? @SebastienGllmt

Reply:
Redeemers don't go in the `add_plutus_datum`. Instead, you can use `add_redeemer`. The partial redeemers need to be combined with an index and a tag. The idea is that we can know the tag and the index so the user doesn't have to pass them in (ex: if they called `add_certificate`, the index is next available position in the certificates array and the tag is "Cert". You should be able to complete these partial redeemers so that they are full redeemers from within the context  of the tx_builder

Reply:
I've used `add_redeemer` in `add_*` of transaction builder instead. Please review.